src/index.coffee

```diff
@@ -9,7 +9,7 @@ module.exports = (env, callback) ->
 
     getPages: (contents) ->
       env.helpers.contents.list(contents).filter (content) ->
-        content instanceof env.plugins.MarkdownPage
+        content instanceof env.plugins.MarkdownPage unless content.metadata?.noindex
```

Comment:
I would rather use `and not` because it ensures the function passed to `filter` always returns a boolean value. If you use `unless`, it does not ensure it as the function may returns `true`, `false` or `undefined`.

In addition, all objects instantiated from `MarkdownPage` (and its children like `JsonPage`) contains a `metadata` property. So, it is safe to write here: `content.metadata.noindex`.